src/bodypart.h

```diff
@@ -107,6 +107,39 @@ struct stat_hp_mods {
 
 struct body_part_type {
     public:
+        /**
+         * the different types of body parts there are.
+         * this allows for the ability to group limbs or determine a limb of a certain type
+         */
+        enum class type {
+            // vital organs
+            vital,
```

Comment:
Having part type called `vital` and a `is_vital` property seems a bit confusing, maybe the type should be called `internal`?